x-pack/plugins/security_solution/common/detection_engine/get_query_filter.ts

```diff
@@ -22,7 +22,7 @@ import { Query, Language, Index, TimestampOverrideOrUndefined } from './schemas/
 export const getQueryFilter = (
   query: Query,
   language: Language,
-  filters: Array<Partial<Filter>>,
+  filters: unknown,
```

Comment:
The source of the problem here is that we don't have validation for the `Filter` type on the server side yet, so in attempting to make the server types more accurate the type of `filters` in RuleTypeParams became `unknown`. Then since we had a cast in here on line 51 already I thought we could just do the unsafe cast once at the end instead of passing in a partial then casting to unknown and then to `Filter[]`.

I think you're right though that we should pass in `Filter[]`. As part of the follow up updating server types I'll revisit this - want to get this merged before it conflicts with other work.